docs/DeveloperGuide.adoc

```diff
@@ -92,155 +93,297 @@ The `UI` component,
+If `ManpowerCountToAdd` is not specified, it is assumed to be the maximum number possible for the event.
+Validity of other command arguments e.g. if `ManpowerCountToAdd` is a positive integer is checked
+by `AutoAllocateCommandParser` and not within the command `AutoAllocateCommand`.
 
-The following sequence diagram shows how the undo operation works:
 
-image::UndoSequenceDiagram.png[]
+**Step 3**. The command calls its own method `AutoAllocateCommand#getManpowerNeededByEvent()` to get the number of employees
+required by the specified event.
 
-NOTE: The lifeline for `UndoCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
 
-The `redo` command does the opposite -- it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the address book to that state.
+**Step 4**. The command calls its own method `AutoAllocateCommand#createAvailableEmployeeListForEvent()` to create a
+filtered list of employees based on the `tagList` and if employee satisfies `Event#isAvailableForEvent()`.
 
+**Step 5**. The command checks if supply (number of employees in filtered list in step 4) exceeds
+demand (number of employees required by event, generated in step 3).
 [NOTE]
-If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone address book states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+If supply < demand, throws exception to user. If supply > demand, employees will be randomly selected instead.
+
+**Step 6**. The command calls `Event#allocateEmployee()` to allocate the chosen employees to the event.
+
+[NOTE]
+For storage purposes, only the `Employee#EmployeeId` is allocated to the event.
+
+**Step 7**. Done.
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
 
-image::UndoRedoState4.png[]
+The following sequence diagram shows how the auto allocation works:
 
-Step 6. The user executes `clear`, which calls `Model#commitAddressBook()`. Since the `currentStatePointer` is not pointing at the end of the `addressBookStateList`, all address book states after the `currentStatePointer` will be purged. We designed it this way because it no longer makes sense to redo the `add n/David ...` command. This is the behavior that most modern desktop applications follow.
+._Sequence Diagram of the AutoAllocate Command_
+image::AutoAllocateSequenceDiagram.png[]
```

Comment:
What is this "a" that been returned by AddressBookParser and AutoAllocateCommandParser?